src/main/java/org/apache/accumulo/access/AccessExpressionImpl.java

```diff
@@ -34,11 +34,6 @@ class AccessExpressionImpl implements AccessExpression {
 
   @Override
   public String getExpression() {
-    var expStr = expressionString.get();
-    if (expStr != null) {
-      return expStr;
-    }
-
     return expressionString.updateAndGet(es -> es == null ? new String(expression, UTF_8) : es);
```

Comment:
Won't this re-set `expressionString` every time `getExpression()` is called?

Reply:
Yea I think it will. Nice catch

Reply:
I guess both checks for null are needed in case the value changes between the first and second readings of `expressionString`